### PR TITLE
Astro 1591 e2e test setup

### PR DESCRIPTION
**Pull request by markacianfrani**
## Brief Description

Migrate to Cypress for E2E Testing

`npm run cy.test` to run all E2E tests
`npm run cy.test.watch` to open cypress in watch mode

## JIRA Link

[ASTRO-1591](https://rocketcom.atlassian.net/browse/ASTRO-1591)

## General Notes
* Adds automated check on new PRs. If your PR breaks an E2E test, it cannot be merged.
* Uses storybook for an isolated testing environment
* All existing Stencil E2E tests have been ported over to use Cypress instead.

## Issues and Limitations
* Kind of slow to run on PRs. We could investigate using Cypress Dashboard to run tests in parallel. But the bottleneck is more on the Netlify deploy side and our build time. 
* Only basic 'it renders' stories were ported over. E2E tests are written in js instead of typescript. Cypress supports it but I don't really see a benefit to using it.
* Pop Up Menu E2E stories still need to be ported, blocked by [PR 69](https://github.com/RocketCommunicationsInc/astro-components-stencil/pull/69)

## Types of changes

-   [x] New feature

## Checklist

-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: src/components/rux-notification/test/rux-notification.e2e.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+describe('Notification', () => {
+    beforeEach(() => {
+        cy.visitStory('components-notification--default-story')
+    })
+    it('renders', async () => {
+        cy.get('rux-notification').should('have.class', 'hydrated')
+    })
+
+    //   it('closes after time is up', async () => {
```
Comment (micahjones13) [inline src/components/rux-notification/test/rux-notification.e2e.js:9]: Could you elaborate more on why we don't include the other tests, and instead just have the basic render tests? If cypress supports these js tests, I feel like the more coverage the better. Maybe I'm misunderstanding something?

File: src/components/rux-tabs/test/rux-tabs.e2e.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+describe('Tab', () => {
+    beforeEach(() => {
+        cy.visitStory('components-tabs--default-story')
```
Comment (micahjones13) [inline src/components/rux-tabs/test/rux-tabs.e2e.js:3]: This one is seemingly opposite - no renders test, but all of the others

File: src/components/rux-tree/test/rux-tree.e2e.ts [hunk 3]
```diff
@@ -1,20 +0,0 @@
...
-    let page!: E2EPage
-
-    beforeEach(async () => {
-        page = await newE2EPage()
-        await page.setContent(`<rux-tree>
-        <rux-tree-node id="first">Level 1</rux-tree-node>
-        <rux-tree-node>Level 2</rux-tree-node>
-      </rux-tree>`)
-    })
-    it('focuses first node', async () => {
```
Comment (micahjones13) [inline src/components/rux-tree/test/rux-tree.e2e.ts:13]: Was this test omitted for any reason?

**Review by micahjones13 — CHANGES_REQUESTED** [review]

File: src/components/rux-notification/test/rux-notification.e2e.js (hunk 1 above)
Comment (markacianfrani) [reply src/components/rux-notification/test/rux-notification.e2e.js:9]: absolutely! we'll definitely be adding more in separate tickets. This PR is just to get the foundation up and running and provide some basic examples.

File: src/components/rux-tree/test/rux-tree.e2e.ts (hunk 3 above)
Comment (markacianfrani) [reply src/components/rux-tree/test/rux-tree.e2e.ts:13]: will be added in a bigger PR that will include more extensive Tree tests

File: src/components/rux-tabs/test/rux-tabs.e2e.js (hunk 2 above)
Comment (markacianfrani) [reply src/components/rux-tabs/test/rux-tabs.e2e.js:3]: added back in

**Review by micahjones13 — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -17,18 +17,22 @@
...
+        "cy.start-storybook": "start-storybook -p 6060 -s ./dist -c .storybook-next",
+        "cy.test": "concurrently \"start-storybook -p 6060 -c .storybook-next -s ./dist\" \"wait-on http-get://localhost:6060\" \"npm run cy.run\" --success first",
+        "cy.test.watch": "concurrently \"npm run start\" \"start-storybook -p 6060 -c .storybook-next -s ./dist\" \"wait-on http-get://localhost:6060\" \"npm run cy.open\" --success first",
         "test.spec": "TZ=UTC stencil test --spec",
-        "test.e2e": "TZ=UTC stencil test --e2e",
         "test.coverage": "TZ=UTC stencil test --spec --coverage",
-        "test.watch": "TZ=UTC stencil test --spec --e2e --watchAll",
-        "test.regression": "npm-run-all --parallel start storybook",
+        "test.watch": "TZ=UTC stencil test --spec --watchAll",
+        "test.regression": "concurrently \"npm run start\" \"npm run storybook\"",
```
Comment (nortonprojects) [inline package.json:29]: Both `concurrently` and `npm-run-all` are used in here, it looks like one of them can be gotten rid of.

**Review by nortonprojects — CHANGES_REQUESTED** [review]
Once `concurrently` or `npm-run-all` are chosen, this will be good.

I prefer to get rid of example files, but that's not something that should hold up the PR at this point

File: cypress/plugins/index.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/// <reference types="cypress" />
```
[the remark this answers is not in the document]
Comment (markacianfrani) [reply cypress/plugins/index.js:1]: nope removed

File: package.json (hunk 4 above)
Comment (markacianfrani) [reply package.json:29]: removed npm run all in favor of concurrently to avoid having to write even more one off npm scripts

**Review by nortonprojects — APPROVED** [review]

**Pull request merged**